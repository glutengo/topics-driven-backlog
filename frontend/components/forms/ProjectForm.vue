<template>
    <BForm
        @submit.prevent="submit"
        class="project-form">

        <BFormGroup 
            label="Title" 
            label-for="project-title">

            <BFormInput 
                ref="focusInput"
                id="project-title" 
                v-model="data.title" 
                required/>
        </BFormGroup>

        <BButton 
            type="submit" 
            variant="primary"
            :class="{ 'is-disabled': isInvalid }">Save</BButton>

        <BButton 
            type="button" 
            variant="grey" 
            @click="cancel">Cancel</BButton>
    </BForm>
</template>

<script>
import { required } from 'vuelidate/lib/validators';
import BForm from '@bootstrap/form/form';
import BFormGroup from '@bootstrap/form-group/form-group';
import BFormInput from '@bootstrap/form-input/form-input';
import BButton from '@bootstrap/button/button';
import BaseForm from '@/components/forms/BaseForm';

export default {
    name: 'ProjectForm',
    components: { BForm, BFormGroup, BFormInput, BButton },
    extends: BaseForm,
    validations: {
        data: {
            title: { required },
        },
    },
};
</script>
