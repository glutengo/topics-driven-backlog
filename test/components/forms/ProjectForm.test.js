import { mount, createLocalVue } from 'vue-test-utils';
import Vuelidate from 'vuelidate';
import ProjectForm from '@/components/forms/ProjectForm';


const localVue = createLocalVue();
localVue.use(Vuelidate);

describe('ProjectForm.test.js', () => {
    let cmp;

    beforeEach(() => {
        cmp = mount(ProjectForm, {
            localVue,
            propsData: {
                data: {
                    title: 'Test Project',
                },
            },
        });
    });

    it('has the expected html structure', () => {
        expect(cmp.element).toMatchSnapshot();
    });
});
