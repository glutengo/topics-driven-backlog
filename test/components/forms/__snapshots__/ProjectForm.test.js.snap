// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectForm.test.js has the expected html structure 1`] = `
<form
  class="project-form"
>
  <div
    aria-labelledby="__BVID__6___BV_label_"
    class="b-form-group form-group"
    id="__BVID__6_"
    role="group"
  >
    <label
      class="col-form-label pt-0"
      for="project-title"
      id="__BVID__6___BV_label_"
    >
      Title
    </label>
    <div
      class=""
    >
      <input
        aria-required="true"
        class="form-control"
        id="project-title"
        required="required"
        type="text"
      />
      <!---->
      <!---->
      <!---->
    </div>
  </div>
   
  <button
    class="btn btn-primary"
    type="submit"
  >
    Save
  </button>
   
  <button
    class="btn btn-grey"
    type="button"
  >
    Cancel
  </button>
</form>
`;
